test/10_before_test/setup.py

```diff
@@ -0,0 +1,19 @@
+from setuptools import setup, Extension
+import sys, os
+
+# assert that the Python version as written to pythonversion.txt in the CIBW_BEFORE_BUILD step
+# is the same one as is currently running.
+version_file = 'c:\\pythonversion.txt' if sys.platform == 'win32' else '/tmp/pythonversion.txt'
+if os.path.exists(version_file):
+    os.remove(version_file)
+
+# check that the executable also was written
+executable_file = 'c:\\pythonexecutable.txt' if sys.platform == 'win32' else '/tmp/pythonexecutable.txt'
+if os.path.exists(executable_file):
+    os.remove(executable_file)
```

Comment:
Just thinking- I'm not sure we need these delete steps here in `setup.py`..?

Reply:
I agree. The reason these were in `CIBW_BEFORE_BUILD` is probably because only `setup.py` gets run here. Now there's a test suite that's run, so things should be checked there.

Reply:
In macos build cibuildwheel use symlink for choosing python versions. Then all python version has same executable path. So we need to delete this files to be sure that `CIBW_BEFORE_BUILD` is executed properly.

Reply:
This is an *excellent* point about `pythonversion.txt`. I previously said that `pythonversion.txt` should be removed in this test - I was wrong, for the symlink reason above, we should keep that. Apologies @Czaki!

However, I still think the `setup.py` 'remove files' step is unnecessary. Each test will overwrite the previous version.

Reply:
I revert this change.